client/Packages/com.beamable/Runtime/Modules/Purchasing/UnityBeamablePurchaser.cs

```diff
@@ -384,6 +391,41 @@ private IEnumerator RetryTransaction(CompletedTransaction transaction, Product p
 		}
 	}
 
+	public struct SkuIdPair
+	{
+		/// <summary>
+		/// The product id from Beamable's SKU content. <see cref="SKU.productIds"/>
+		/// </summary>
+		public string skuProductId;
+		
+		/// <summary>
+		/// The IAP store id
+		/// </summary>
+		public string storeId;
+	}
+	
+	public class UnityBeamablePurchaserUtil
+	{
+		/// <summary>
+		/// Create a <see cref="IDs"/> type from a list of Beamable SKU product id and IAP store ids.
+		/// This method will only add the beamable SKU product ids that are not null or empty strings.
+		/// </summary>
+		/// <param name="skuIdPairs"></param>
+		/// <returns></returns>
+		public static IDs CreateIdsFromSku(params SkuIdPair[] skuIdPairs)
+		{
+			var ids = new IDs();
+
+			foreach (var pair in skuIdPairs)
+			{
+				if (string.IsNullOrEmpty(pair.skuProductId)) continue;

```

Comment:
I feel like this filtering should happen before we even get here. That is, rather than "for every SKU, build a list of platform-specific product IDs, skipping those which are null," I would have an easier time remembering, understanding, and therefore maintaining, "for the platform(s) relevant in this context, gather all SKUs that have a product ID for that platform, and add them to the builder." Not a _huge_ distinction, but my 2c about simplicity of logic (in the interest of maintainability).

Reply:
Hashed it out with @cdhanna in Slack and came to the conclusion that my biggest gripe was how far away, lexically, the null-filtering function was (line 400-ish) from the place where the filtering mattered (line 70-ish). The inline function is a nice compromise that satisfies the "it is too far away" complaint without caving in to my absurd stance of, "why DRY? it would look better with repetition" 😄